<a href="https://colab.research.google.com/github/lethikieuly/PANDAS-FUNCTION/blob/main/Pandas_Size_shape_ndim.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Python is a great language for doing data analysis, primarily because of the fantastic ecosystem of data-centric Python packages. Pandas is one of those packages and makes importing and analyzing data much easier.

Pandas .size, .shape and .ndim are used to return size, shape and dimensions of data frames and series.


1.   dataframe.size: returns size of dataframe. Ví dụ: một mảng [1,2][2,4] thì size sẽ bằng số hàng* số cột = 4

2.   dataframe.shape: Tuple of array dimensions. Tiếp  tục ví dụ trên shape =(2,2)

3.   dataframe.ndim: Returns dimension of dataframe/series. 1 for one dimension (series), 2 for two dimension (dataframe)



In [1]:
import pandas as pd

In [2]:
data =pd.read_csv("/content/sample_data/nba.csv")

In [3]:
data.head(5)

Name            Team  Number  ... Weight            College     Salary
0  Avery Bradley  Boston Celtics     0.0  ...  180.0              Texas  7730337.0
1    Jae Crowder  Boston Celtics    99.0  ...  235.0          Marquette  6796117.0
2   John Holland  Boston Celtics    30.0  ...  205.0  Boston University        NaN
3    R.J. Hunter  Boston Celtics    28.0  ...  185.0      Georgia State  1148640.0
4  Jonas Jerebko  Boston Celtics     8.0  ...  231.0                NaN  5000000.0

[5 rows x 9 columns]

In [8]:
size = data.size 
shape = data.shape
df_ndim=data.ndim
series_ndim= data["Salary"].ndim
print("Size={}\nShape= {}".format(size,shape))
print("ndim of dataframe = {}\nndim of series ={}".format(df_ndim, series_ndim))

Size=4122
Shape= (458, 9)
ndim of dataframe = 2
ndim of series =1


Từ kết quả có size = 458*9 và Ngoài ra, ndim cho khung dữ liệu là 2 và chuỗi là 1, đúng cho tất cả các loại khung dữ liệu và chuỗi.